File: Tests Objective-C/RTMPConnectionTests.m
#import <XCTest/XCTest.h>
#import <lf/lf-Swift.h>

@interface RTMPConnectionTests : XCTestCase

@end

@implementation RTMPConnectionTests

- (void)testConnect {
    RTMPConnection *conn = [[RTMPConnection alloc] init];
    [conn connect: @"rtmp://localhost:8080"];
}

@end

File: Tests/RTMP/RTMPConnectionTests.swift
import Foundation
import XCTest

@testable import HaishinKit

final class RTMPConnectionTests: XCTestCase {
    func publish() {
        let connection: RTMPConnection = RTMPConnection()
        let stream: RTMPStream = RTMPStream(connection: connection)
        connection.connect("rtmp://localhost:1935/live")
        stream.publish("live")
        sleep(10000)
    }

    func testReleaseWhenClose() {
        weak var weakConnection: RTMPConnection?
        _ = {
            let connection = RTMPConnection()
            connection.connect("rtmp://localhost:1935/live")
            connection.close()
            weakConnection = connection
        }()
        XCTAssertNil(weakConnection)
    }
}
